### PR TITLE
[Log Collector] Add HasLogs endpoint + additional fixes

### PR DESCRIPTION
**Pull request by TomerShor**
In this PR there are several additions and fixes wrt the log collector gRPC service:

- Add `HasLogs` endpoint in the log collector server, for the api to call before calling `get_logs`. This enables the api to gracefully return empty logs to the client if the run has no logs to retrieve, instead of exploding.
- Align env vars according to https://github.com/v3io/helm-charts/pull/913.
- Limit the log collector to monitor goroutines, update the state file and call `StartLogs` for runs in progress only if it is in the chief pod (according the the env var `MLRUN_HTTPDB__CLUSTERIZATION__ROLE`.
- Support error codes in the grpc responses. Currently supports `NotFound` and `InternalError`.
- Fix log files and state file paths.
- Log stack trace in case of unexpected internal errors.

### CODE REVIEW

File: go/pkg/services/logcollector/server.go [hunk 1]
```diff
@@ -331,6 +340,33 @@ func (s *Server) GetLogs(request *protologcollector.GetLogsRequest, responseStre
 	return nil
 }
 
+// HasLogs returns true if the log file exists for a given run id
+func (s *Server) HasLogs(ctx context.Context, request *protologcollector.HasLogsRequest) (*protologcollector.HasLogsResponse, error) {
+
+	s.Logger.DebugWithCtx(ctx, "Received Has Logs request", "runUID", request.RunUID)
```
Comment (liranbg) [inline go/pkg/services/logcollector/server.go:346]: spammy

File: go/pkg/services/logcollector/server.go [hunk 2]
```diff
@@ -331,6 +340,33 @@ func (s *Server) GetLogs(request *protologcollector.GetLogsRequest, responseStre
...
+
+	// get log file path
+	_, err := s.getLogFilePath(ctx, request.RunUID, request.ProjectName)
+	if err != nil {
```
Comment (liranbg) [inline go/pkg/services/logcollector/server.go:350]:
```suggestion
	if _, err := s.getLogFilePath(ctx, request.RunUID, request.ProjectName); err != nil {
```

File: go/pkg/services/logcollector/server.go [hunk 3]
```diff
@@ -331,6 +340,33 @@ func (s *Server) GetLogs(request *protologcollector.GetLogsRequest, responseStre
...
+		if strings.Contains(common.GetErrorStack(err, 10), "not found") {
```
Comment (liranbg) [inline go/pkg/services/logcollector/server.go:351]: why not checking the `errors.RootCause(err).Is(os.FileNotFoundErr)` or something similar

**Review by liranbg — CHANGES_REQUESTED** [review]
minor

File: go/pkg/services/logcollector/server.go (hunk 3 above)
Comment (TomerShor) [reply go/pkg/services/logcollector/server.go:351]: It is a custom search for the file, and hence a custom error.
I don't want to return an `os` error, but I can create a custom `NotFound` error maybe.
I agree about the `RootCasue` tho.

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**